pybamm/solvers/base_solver.py

```diff
@@ -315,6 +317,32 @@ def report(string):
                                 pybamm.EventType.DISCONTINUITY,
                             )
                         )
+                elif isinstance(symbol, pybamm.Modulo):
+                    found_t = False
+                    # Dimensionless
+                    if symbol.left.id == pybamm.t.id:
+                        expr = symbol.right
+                        found_t = True
+                    # Dimensional
+                    elif symbol.left.id == (pybamm.t * model.timescale).id:
+                        expr = symbol.right.new_copy() / symbol.left.right.new_copy()
+                        found_t = True
+
+                    # Update the events if the modulo function depended on t
+                    if found_t:
+                        if t_eval is None:
+                            N_events = 200
+                        else:
+                            N_events = t_eval[-1] // expr.new_copy().value
```

Comment:
I don't think you need a `new_copy` here (if just taking the value)

Reply:
Done